base_layer/key_manager/src/cipher_seed.rs

```diff
@@ -141,10 +140,11 @@ impl CipherSeed {
         CipherSeed::new_with_birthday(birthday)
     }
 
+    /// Generate a new seed with a given birthday
     fn new_with_birthday(birthday: u16) -> Self {
-        let mut entropy = [0u8; CIPHER_SEED_ENTROPY_BYTES];
-        OsRng.fill_bytes(&mut entropy);
-        let mut salt = [0u8; CIPHER_SEED_SALT_BYTES];
+        let mut entropy = vec![0u8; CIPHER_SEED_ENTROPY_BYTES];
```

Comment:
Does this need a `Zeroizing` wrapper since it's a vector and the eventual initialized `CipherSeed` struct zeroizes on drop?

Reply:
If so, it would need to be dereferenced and cloned (see other comment).

Reply:
No the vector (pointer + length) will be moved into the CipherSeed struct, and the underlying buffer is zeroized.

```rust
    #[test]
    fn zerioze() {
        let mut seed = CipherSeed::new();
        unsafe {
            let ptr = seed.entropy.as_ptr();
            seed.zeroize();
            for i in 0..(CIPHER_SEED_ENTROPY_BYTES as isize) {
                assert_eq!(*(ptr.offset(i)), 0, "not zero at index {}", i);
            }
        }
    }
```